tests/manage/cluster/test_must_gather.py

```diff
@@ -0,0 +1,181 @@
+import os
+import logging
+
+import pytest
+
+from ocs_ci.framework.testlib import ManageTest, tier1, bugzilla
+from ocs_ci.ocs import openshift_ops, ocp
+from ocs_ci.ocs.utils import collect_ocs_logs
+from ocs_ci.utility.utils import ocsci_log_path
+
+logger = logging.getLogger(__name__)
+
+
+@tier1
+@pytest.mark.polarion_id("OCS-1583")
+@bugzilla('1766646')
+class TestMustGather(ManageTest):
+
+    @pytest.fixture(autouse=True)
+    def init_ocp(self):
+        """
+        init OCP() object
+        """
+
+        self.ocs = openshift_ops.OCP()
+        self.ocp_obj = ocp.OCP()
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+
+        def finalizer():
+            must_gather_pods = self.ocs.get_pods(label_selector='app=must-gather')
+            logger.info(f"must_gather_pods: {must_gather_pods} ")
+
+            while must_gather_pods:
```

Comment:
this could lead to an infinite loop.
Please use the TimeoutSampler form instead of while loop